From b867b40db382c23586dc89dff1ef4db855538f17 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 01 Apr 2020 20:30:06 +0300
Subject: [PATCH] [SW-0] spi: print protection status only once

Current code prints the protection status whenever the spi_copy()
function is called.
This bloats the serial log and slowers boot (although not
significantly).
Introduce a flag which will takes care to print only once.

Change-Id: I25bff79d9b32fe13d992374c04184005a9275b52
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/spi.c b/common/spi.c
index 44e057b..6033594 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -237,15 +237,18 @@
 void spi_copy(void *dst, unsigned int offset, unsigned int size, char *name)
 {
 	unsigned char sr;
+	static bool print_wp_status = true;
 
 	spi_init();
 	spi_read(dst, offset, size);
 	sr = spi_prot();
 	preboot_puts_raw(name);
 	preboot_puts(" copying from flash is completed\n");
-	if ((sr & PROT_MASK) == PROT_MASK)
-		preboot_puts("Flash may be protected, depend on WP# signal\n");
-	else
-		preboot_puts("Flash is not protected\n");
-
+	if (print_wp_status) {
+		print_wp_status = false;
+		if ((sr & PROT_MASK) == PROT_MASK)
+			preboot_puts("Flash may be protected, depend on WP# signal\n");
+		else
+			preboot_puts("Flash is not protected\n");
+	}
 }
